examples/water-model/water-model.py

```diff
@@ -330,320 +354,269 @@
+p3m_calculator = torchpme.metatensor.P3MCalculator(
+                potential=torchpme.CoulombPotential(p3m_smearing),
+                **p3m_parameters
+            )
 
 # %%
-# Analyze the Results
-# -------------------
-#
-# Time Evolution of Physical Properties
-# ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
-#
-# We examine the time evolution of physical properties. For better comparison, an
-# arbitrary constant is subtracted from the potential energy.
-
-fig, ax = plt.subplots(3, figsize=(8, 5), sharex=True)
-
-time = 0.5 * np.arange(n_steps)
-
-ax[0].plot(time, potential_energy - potential_energy[-1], label="Potential Energy")
-ax[0].plot(time, kinetic_energy, label="Kinetic Energy")
-ax[0].plot(time, total_energy - potential_energy[-1], label="Total Energy")
-ax[0].legend(ncol=3)
-ax[0].set_ylabel("Energy [eV]")
-
-ax[1].plot(time, temperature, label="Temperature")
-ax[1].axhline(300, color="black", linestyle="--", label="Target Temperature")
-ax[1].legend(ncol=2)
-ax[1].set_ylabel("Temperature [K]")
-
-ax[2].plot(time, pressure)
-ax[2].axhline(0, color="black", linestyle="--")
-ax[2].set_ylabel("Pressure [eV Å$^{-3}$]")
-
-ax[-1].set_xlabel("Time [fs]")
-
-fig.align_labels()
-plt.show()
+p3m_calculator.forward(m_sys)
 
 # %%
-#
-# We find that temperature remains around the target of 300 K, and the pressure
-# stabilizes around a positive value.
-#
-# Radial Distribution Function
-# ^^^^^^^^^^^^^^^^^^^^^^^^^^^
-#
-# We compute and plot the partial radial distribution functions (RDF) between all pairs
-# in the system, using the last 20 steps to ensure at least some equilibration from the
-# initial configuration.
-
-OO_rdf_l = []
-OH_rdf_l = []
-HH_rdf_l = []
-
-for atoms in trajectory[-20:]:
-    OO_rdf_step, rdf_dists = ase.geometry.rdf.get_rdf(
-        atoms, rmax=6.0, nbins=200, elements=(8, 8)
-    )
-    OH_rdf_step, _ = ase.geometry.rdf.get_rdf(
-        atoms, rmax=6.0, nbins=200, elements=(8, 1)
-    )
-    HH_rdf_step, _ = ase.geometry.rdf.get_rdf(
-        atoms, rmax=6.0, nbins=200, elements=(1, 1)
-    )
-
-    OO_rdf_l.append(OO_rdf_step)
-    OH_rdf_l.append(OH_rdf_step)
-    HH_rdf_l.append(HH_rdf_step)
+class QTIP4PfModel(torch.nn.Module):
+    def __init__(self, 
+                cutoff: float,
+                lj_sigma: float,
+                lj_epsilon: float,
+                m_gamma: float,
+                m_charge: float,
+                oh_bond_eq: float,
+                oh_bond_d: float,
+                oh_bond_alpha: float,
+                hoh_angle_eq: float,
+                hoh_angle_k: float,
+                p3m_options: Optional[dict]=None,
+                dtype: Optional[torch.dtype]=None,
+                device: Optional[torch.device]=None,
+    ):
+        super().__init__()
+
+        if p3m_options is None:
+            # sane defaults, should give a ~1e-4 relative accuracy on forces 
+            p3m_options = (1.4, {'interpolation_nodes': 5, 'mesh_spacing': 1.33}, 0)
```

Comment:
I would make them at least depend on the `cutoff`.

Reply:
then we should have heuristics in torch-pme. I don't think this is the place to introduce those heuristics